guides/source/action_controller_response_and_rendering.md

```diff
@@ -0,0 +1,703 @@
+
+[`redirect_back`]: https://api.rubyonrails.org/classes/ActionController/Redirecting.html#method-i-redirect_back
+
+### Getting a Different Redirect Status Code
+
+Rails uses HTTP status code 302, a temporary redirect, when you call `redirect_to`. If you'd like to set a different status code, perhaps 301, a permanent redirect, you can use the `:status` option:
+
+```ruby
+redirect_to books_path, status: 301
+```
+
+Just like the `:status` option for `render`, `:status` for `redirect_to` accepts both numeric and symbolic values.
+
+The Difference Between `render` and `redirect_to`
+------------------------------------------------
+
+As we have seen above, the `redirect_to` method is used to send an HTTP redirect response instructing the client's browser to make a new request to a different URL. And the `render` method is used to render a view template from the current action. As we saw, `render` has many options that allow it to render views from different actions or even different controllers.
+
+So what is the difference between `render` and `redirect_to`? Consider this example:
+
+```ruby
+# This code does not work as expected.
+def index
+  @books = Book.all
+end
+
+def show
+  @book = Book.find_by(id: params[:id])
+  if @book.nil?
+    render action: "index"
+  end
+end
+```
+
+In the `show` action, if a book with a given `id` is not found, we attempt to show the user all the books by rendering the `index` action with `render action: "index"`. This does not make sense. Here's why:
+
+The `render` call will find the `index.html.erb` file and attempt to render it. However, the `index` view expects a `@books` instance variable to be set. Since we are coming from the `show` action, while trying to render the `index` view, the `@books` instance variable will *not* be set. Note that `render action: "index"` does *not* execute the code in the `index` controller action before rendering the view, it just attempts to render the view specified.
+
+This is the difference between `render` and `redirect_to`. The `redirect_to` method ends the current action, the browser makes a brand new HTTP request, which in turn, execute the appropriate controller action.
+
+The above example updated to use `redirect_to` instead of `render`:
+
+```ruby
+# This code will work. But is not typically what we do.
+def index
+  @books = Book.all
+end
+
+def show
+  @book = Book.find_by(id: params[:id])
+  if @book.nil?
+    redirect_to action: :index
+  end
+end
+```
+
+With this code, the browser will make a new request for the index page, the code in the `index` method will run, which will set the `@books` instance variable, then the `index.html.erb` view will be rendered. And it will all work as expected.
+
+Note that the above example is for demonstration. It is not typical to render all books from a `show` action for a given book. If a resource is not found, it is reasonable to return a HTTP 404 Not Found status code when `@book.nil` is true.
```

Comment:
If the example is not typical, should we use a more typical example like a `create` action instead?
This example could also explain that a redirect prevents a double submit when using the back button after a form is submitted?